lib/event_sourcery/aggregate_root.rb

```diff
@@ -124,8 +124,12 @@ def load_history(events)
 
     def apply_event(event_class, options = {})
       event = event_class.new(**options.merge(aggregate_id: id))
-      mutate_state_from(event)
-      @changes << event
+      if event.valid?
```

Comment:
Do I understand the sequence of things here correctly?

1) A command handler will call a method on an aggregate root
2) The aggregate root will typically call `apply_event` with some data
3) The event gets instantiated
4) The state of the aggregate gets mutated
5) The event gets saved to the event store (where is that code?)

My concern is if we raise the validation error at (5), the aggregate root instance that the command handler knows about will have already had its state mutated.

Perhaps that doesn't matter because the command would fail anyway.